rest/route/host.go

```diff
@@ -314,6 +316,133 @@ func getLimit(vals url.Values) (int, error) {
 	return limit, nil
 }
 
+////////////////////////////////////////////////////////////////////////
+//
+// POST /users/{user_id}/clear_hosts
+
+func makeClearHostsByUser(sc data.Connector, env evergreen.Environment) gimlet.RouteHandler {
+	return &clearHostsHandler{
+		sc:  sc,
+		env: env,
+	}
+}
+
+type clearHostsHandler struct {
+	user   string
+	dryRun bool
+
+	env evergreen.Environment
+	sc  data.Connector
+}
+
+func (ch clearHostsHandler) Factory() gimlet.RouteHandler {
+	return &clearHostsHandler{
+		sc:  ch.sc,
+		env: ch.env,
+	}
+}
+
+func (ch *clearHostsHandler) Parse(ctx context.Context, r *http.Request) error {
+	ch.user = gimlet.GetVars(r)["user_id"]
+	if ch.user == "" {
+		return gimlet.ErrorResponse{
+			Message:    "user cannot be empty",
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+	vals := r.URL.Query()
+	ch.dryRun = vals.Get("dry_run") == "true"
+
+	return nil
+}
+
+func (ch *clearHostsHandler) Run(ctx context.Context) gimlet.Responder {
+	usr := gimlet.GetUser(ctx)
+	opts := model.APIHostParams{
+		UserSpawned: true,
+	}
+	// returns all up-hosts for user
+	hosts, err := ch.sc.FindHostsInRange(opts, ch.user)
+	if err != nil {
+		return gimlet.NewJSONErrorResponse(errors.Wrap(err, "database error getting hosts"))
+	}
+
+	volumes, err := ch.sc.FindVolumesByUser(ch.user)
+	if err != nil {
+		return gimlet.NewJSONErrorResponse(errors.Wrap(err, "database error getting volumes"))
+	}
+
+	toTerminate := model.APIClearHostsResults{
+		Hosts:   []string{},
+		Volumes: []string{},
+	}
+	currentTime := time.Now()
+
+	mgrCache := map[cloud.ManagerOpts]cloud.Manager{}
+	for _, v := range volumes {
+		if !v.NoExpiration && v.Expiration.Before(currentTime) { // already terminated
```

Comment:
don't volumes get removed from the database when they're [deleted](https://github.com/evergreen-ci/evergreen/blob/c12ad31b5a22c76bc36073c9bb8c09c82391cf43/cloud/ec2.go#L1332)?

Reply:
Theoretically yes but that was added late--I know I currently have volume documents for volumes that don't exist anymore. This seemed like a good check just in case. 

I looked it up and there are 17 volumes that don't have Expiration or NoExpiration set, which i think are dead documents, so I could go through and just delete them and then remove this check.

Reply:
sgtm.
(My guess would be that they're from before expiration was implemented, so the reaper got them)

Reply:
Sorry, which thing sounds good, leaving the check in or deleting the volumes?

Reply:
_Actually_ since DeleteVolume doesn't error if the instance is found we don't need this check either way.

Reply:
I meant deleting the volumes (it would be good to clean them up at some point. We shouldn't be generating any new ones now that expiration is a thing), but if you prefer leaving the check that's okay with me too.